OpenRA.Mods.Common/Traits/Targetable.cs

```diff
@@ -29,24 +30,23 @@ public class TargetableInfo : UpgradableTraitInfo, ITargetableInfo
 	public class Targetable : UpgradableTrait<TargetableInfo>, ITargetable, INotifyCreated
 	{
 		protected static readonly string[] None = new string[] { };
-		protected Cloak cloak;
+		protected Cloak[] cloak;
 
 		public Targetable(Actor self, TargetableInfo info)
 			: base(info) { }
 
 		void INotifyCreated.Created(Actor self)
 		{
-			cloak = self.TraitOrDefault<Cloak>();
+			cloak = self.TraitsImplementing<Cloak>().ToArray();
 		}
 
 		public virtual bool TargetableBy(Actor self, Actor viewer)
 		{
 			if (IsTraitDisabled)
 				return false;
-			if (cloak == null || (!viewer.IsDead && viewer.Info.HasTraitInfo<IgnoresCloakInfo>()))
```

Comment:
For safety (.IsVisible may do other trait lookups) I would keep this check here, but replace the null check with the .Any() check.